openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -108,11 +111,27 @@ func CreateCinderVolume(vs *gophercloud.ServiceClient, options controller.Volume
 // become available.  The connection information cannot be retrieved from cinder
 // until the volume is available.
 func WaitForAvailableCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
-	// TODO: Implement proper polling instead of brain-dead timers
 	c := make(chan error)
-	go time.AfterFunc(5*time.Second, func() {
-		c <- nil
-	})
+	go func() error {
+		num := 0
+		for {
+			time.Sleep(sleepTime * time.Second)
+			if num > maxRetry {
+				err := errors.New("Failed to get volume status after max retries")
```

Comment:
This error would be better written as: "Volume not available after max retries"

Reply:
Done in next change